### PR TITLE
do not allow out of bounds access to the EnumName##_Strings array

### PR DESCRIPTION
**Pull request by anporumb**

### CODE REVIEW

File: inc/testrunnerswitcher.h [hunk 1]
```diff
@@ -62,10 +62,17 @@ static const wchar_t *EnumName##_Strings[]= \
...
+{                                                                                                                                   \
+    if(enumValue<(sizeof(EnumName##_Strings)/sizeof(EnumName##_Strings[0])))                                                        \
+    {                                                                                                                               \
+        (void)snprintf(dest, bufferSize, "%ls", EnumName##_Strings[enumValue]);                                                     \
+    }                                                                                                                               \
+    else                                                                                                                            \
+    {                                                                                                                               \
+        (void)snprintf(dest, bufferSize, C3("%d is out of bounds value for ", MU_TOSTRING(EnumName), ""), enumValue);               \
```
Comment (dcristoloveanu) [inline inc/testrunnerswitcher.h]: >C3( [](start = 41, length = 3)

I don't think C3 is actually needed here, is it?

Comment (dcristoloveanu) [reply inc/testrunnerswitcher.h]: Thus if not needed, I'd say remove it :-).

---
In reply to: [292239776](https://github.com/Azure/azure-c-testrunnerswitcher/pull/20#discussion_r292239776) [](ancestors = 292239776)

Comment (anporumb) [reply inc/testrunnerswitcher.h]: It is not, you are right. Updated

---
In reply to: [292239840](https://github.com/Azure/azure-c-testrunnerswitcher/pull/20#discussion_r292239840) [](ancestors = 292239840,292239776)

Comment (anporumb) [reply inc/testrunnerswitcher.h]: something happened to github and it doesn't see my latest pushes :(((((((

---
In reply to: [292653312](https://github.com/Azure/azure-c-testrunnerswitcher/pull/20#discussion_r292653312) [](ancestors = 292653312,292239840,292239776)

**Pull request closed without merging**